test/quadiron_c_utest.cpp

```diff
@@ -213,6 +197,39 @@ class QuadironCTest : public ::testing::Test {
             }
         }
 
+        for (int i = 0; i < n_data; i++) {
+            if (missing_idxs[i]) {
+                std::fill_n(_data[i], block_size + metadata_size, 0);
+            }
+        }
+
+        for (int i = 0; i < n_parities; i++) {
+            if (missing_idxs[n_data + i]) {
+                std::fill_n(_parity[i], block_size + metadata_size, 0);
+            }
+        }
+
+        // FIXME: for non-systematic FNT, `quadiron_fnt32_decode` will
```

Comment:
IMHO, it's cleaner to have a copy of `_data`/call `quadiron_fnt32_decode` on a copy instead of relying on the order of the test.

Reply:
yes, it's better. A copy of encoded fragments will be used to avoid such cases.